### DISCUSSION TITLE
When to add changelog?

### DISCUSSION THREAD

**Discussion by alamenai** [olivierlacan/keep-a-changelog#504] [Q&A]
It seems not the right place for this question, but I guess the right community to answer it.

I'm just confusing if the changelog will be from the first ( initial ) release or only after making a patch, minor or major changes.

Let's say I published the first package 0.1.0, should the repository contains a changelog about this release?

**Comment by Levi-Lesches** [comment]
A changelog is documenting what changes, so I would not count the initial release, since there is nothing to compare it to.

**Comment by booch** [comment]
It might be interesting, as a historical record of what features got added when. (Remember, some changelog files are named `HISTORY`.) I'm doing that for a side project I just started. But it's not strictly necessary for the users of my app, because there are none yet.

**Comment by plocket** [comment]
In my current project, we only started a change log with the first stable version.

Right now it looks to me like changelogs are useful to people who want to know what behavior to expect from the code. Telling people about behavior that they shouldn't rely on seemed like it would be unhelpful and confusing.
